nucliadb_node/src/services/writer.rs

```diff
@@ -371,7 +371,21 @@ impl ShardWriterService {
         relation_result?;
         Ok(())
     }
-
+    pub fn list_vectorsets(&self) -> InternalResult<Vec<String>> {
+        let reader = self.vector_writer.read().unwrap();
```

Comment:
```rust
self.vector_writer.read().unwrap_or_else(PoisonError::into_inner)
```

Reply:
I think the node should panic if there is a PoisonError. Why you think it should  throw that error to the caller?

Reply:
It was more a suggestion that a strong opinion of mine. But definitely, unwrapping here is not the way to go. If you think that any panic if a thread should be propagated, then return a result. If not, just throw the poison error and acquire the lock

Reply:
Lets do this in a future iteration, this PR is already to complex. And we would need to look carefully at how to treat this case.